src/mscorlib/shared/System/Diagnostics/Tracing/EventSource.cs

```diff
@@ -1040,7 +1040,7 @@ protected internal struct EventData
             /// Address where the one argument lives (if this points to managed memory you must ensure the
             /// managed object is pinned.
             /// </summary>
-            public IntPtr DataPointer { get { return (IntPtr)m_Ptr; } set { m_Ptr = unchecked((long)value); } }
+            public IntPtr DataPointer { get { return (IntPtr)m_Ptr; } set { m_Ptr = unchecked((ulong)value); } }
```

Comment:
Ack.   You are correct.   Good catch.     I had performed an experiment much like your own but was not paying enough attention and was running 64 bit when I did the experiment (were it worked).   I will open another PR to fix this and more carefully validate it.